tests/integrationv2/test_external_psk.py

```diff
@@ -0,0 +1,183 @@
+                      'S2N_PSK_HMAC_SHA256'
+
+s2n_client_only_psk = 's2n_client_psk_identity,'\
+                      'aea617646faaea6d,'\
+                      'S2N_PSK_HMAC_SHA384'\
+
+s2n_server_only_psk = 's2n_server_psk_identity,'\
+                      'a6a4dafcd0fc67d2a,'\
+                      'S2N_PSK_HMAC_SHA256'\
+
+s2n_invalid_hmac_psk = 'psk_identity,'\
+                       'aea617646faaea6d,'\
+                       'S2N_PSK_HMAC_SHA512'\
+
+S2N_CLIENT_PSK_PARAMETERS = [ 
+    [ '--psk', s2n_client_only_psk ],
+    [ '--psk', s2n_known_value_psk ], 
+    [ '--psk', s2n_invalid_hmac_psk ],
+    [ '--psk', s2n_client_only_psk, '--psk', s2n_known_value_psk ], 
+]
+
+S2N_SERVER_PSK_PARAMETERS = [
+    [ '--psk', s2n_server_only_psk ],
+    [ '--psk', s2n_known_value_psk ],
+    [ '--psk', s2n_invalid_hmac_psk ],
+    [ '--psk', s2n_server_only_psk, '--psk', s2n_known_value_psk ], 
+]
+
+OPENSSL_SERVER_PSK_PARAMETER = [ '-psk_identity', shared_psk_identity, '--psk', shared_psk_secret, '-nocert' ]
+OPENSSL_CLIENT_PSK_PARAMETER = [ '-psk_identity', shared_psk_identity, '--psk', shared_psk_secret ]
+
+def s2n_assert_chosen_psk(idx, results):
+    if idx == 0:
+        assert b"Chosen PSK wire index" not in results.stdout
+        assert results.exit_code != 0
+    elif idx % 2 == 1:
+        assert b"Chosen PSK type: S2N_PSK_TYPE_EXTERNAL" in results.stdout
+        assert bytes("Chosen PSK identity: {}".format(shared_psk_identity).encode('utf-8')) in results.stdout
+        assert results.exit_code == 0
+    elif idx == 2:
+        assert b"Invalid psk hmac algorithm" in results.stderr
+        assert results.exit_code != 0
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("client_psk_params", S2N_CLIENT_PSK_PARAMETERS, ids=get_parameter_name)
+def test_external_psk_s2nc_with_s2nd(managed_process, client_psk_params):
+    port = next(available_ports)
+
+    random_bytes = data_bytes(64)
+    client_options = ProviderOptions(
+        mode=S2N.ClientMode,
+        host="localhost",
+        port=port,
+        cipher=Ciphers.CHACHA20_POLY1305_SHA256,
+        data_to_send=random_bytes,
+        insecure=True,
+        extra_flags=client_psk_params,
+        protocol=Protocols.TLS13)
+    
+    idx = S2N_CLIENT_PSK_PARAMETERS.index(client_psk_params)
+    server_psk_params = S2N_SERVER_PSK_PARAMETERS[idx]
+
+    server_options = copy.copy(client_options)
+    server_options.data_to_send = None
+    server_options.mode = S2N.ServerMode
+    server_options.extra_flags = server_psk_params
+
+    server = managed_process(S2N, server_options, timeout=20)
+    client = managed_process(S2N, client_options, timeout=20)
+
+    for results in client.get_results():
+        s2n_assert_chosen_psk(idx, results)
+        assert results.exception is None
+
+    for results in server.get_results():
+        s2n_assert_chosen_psk(idx, results)
```

Comment:
Shouldn't there be some assertion in the successful cases that we can read the random_data_bytes sent over?